### PR TITLE
Add IBMi Software License field to instance data source and resource

### PR DESCRIPTION
**Pull request by ismirlia**
This PR adds the software license fields to instance data source and resource. These fields can be updated. I have added a new acceptance test.

```
--- PASS: TestAccIBMPIInstanceIBMiLicense (1869.89s)
PASS
```

### CODE REVIEW

File: website/docs/d/pi_instance.html.markdown [hunk 1]
```diff
@@ -84,3 +84,9 @@ In addition to all argument reference list, you can access the following attribu
 - `storage_type` - (String) The storage type where server is deployed.
 - `virtual_cores_assigned` - (Integer) The virtual cores that are assigned to the instance.
 - `volumes` - (List of strings) The list of volume IDs that are attached to the instance.
+
+  **Notes** Ibmi software licenses for IBMi virtual server instances --software licenses currently only supports IBMi instances
```
Comment (michaelkad) [inline website/docs/d/pi_instance.html.markdown:88]:
```suggestion
  **Notes** Ibmi software licenses for IBMi virtual server instances -- only for IBMi instances
```

File: website/docs/r/pi_instance.html.markdown [hunk 2]
```diff
@@ -66,6 +66,10 @@ Review the argument references that you can specify for your resource.
 - `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
 - `pi_deployment_type` - (Optional, String) Custom deployment type; Allowable value: `EPIC`.
 - `pi_health_status` - (Optional, String) Specifies if Terraform should poll for the health status to be `OK` or `WARNING`. The default value is `OK`.
+ **Notes** Ibmi software licenses for IBMi virtual server instances --software licenses currently only supports IBMi instances. Default to `false` and `0` if no values provided
```
Comment (michaelkad) [inline website/docs/r/pi_instance.html.markdown:69]:
```suggestion
 **Notes** Ibmi software licenses for IBMi virtual server instances --only for IBMi instances. Default to `false` and `0` if no values provided
```

**Review by michaelkad — COMMENTED** [review]
Can you update the documentation.  Since those will only be in ibmi.

File: website/docs/r/pi_instance.html.markdown (hunk 2 above)
Comment (ismirlia) [reply website/docs/r/pi_instance.html.markdown]: Made changes

**Comment by michaelkad** [comment]
@ismirlia could you fix the `ibm/service/power/ibm_pi_constants.go` conflict?

**Comment by hkantare** [comment]
fix conflicts

File: ibm/service/power/data_source_ibm_pi_instance.go [hunk 3]
```diff
@@ -162,6 +162,26 @@ func DataSourceIBMPIInstance() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			PIInstanceIbmiCSS: {
```
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_instance.go:165]: nit:
```suggestion
			PIInstanceIBMiCSS: {
```

File: ibm/service/power/resource_ibm_pi_network.go [hunk 4]
```diff
@@ -79,7 +79,7 @@ func ResourceIBMPINetwork() *schema.Resource {
 				Optional:      true,
 				Computed:      true,
 				Deprecated:    "This field is deprecated, use pi_network_mtu instead.",
-				ConflictsWith: []string{helpers.PINetworkMtu},
+				ConflictsWith: []string{helpers.PINetworkMtu, helpers.PINetworkJumbo},
```
Comment (yussufsh) [inline ibm/service/power/resource_ibm_pi_network.go:82]: Trying to add the same argument?

File: ibm/service/power/data_source_ibm_pi_instance.go (hunk 3 above)
Comment (yussufsh) [inline ibm/service/power/data_source_ibm_pi_instance.go:165]: I think computed attributes should not start with `pi_`.

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: ibm/service/power/resource_ibm_pi_network.go (hunk 4 above)
Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_network.go:82]: Ah sorry I think this was a typo from the merge conflict.

File: ibm/service/power/data_source_ibm_pi_instance.go (hunk 3 above)
Comment (ismirlia) [reply ibm/service/power/data_source_ibm_pi_instance.go:165]: I don't necessarily disagree, but a lot of the computed attributes in the resource instance already start with pi_.

Comment (ismirlia) [reply ibm/service/power/data_source_ibm_pi_instance.go:165]: sure i'll restyle them

File: website/docs/d/pi_instance.html.markdown [hunk 5]
```diff
@@ -75,3 +75,9 @@ In addition to all argument reference list, you can access the following attribu
 - `storage_type` - (String) The storage type where server is deployed.
 - `virtual_cores_assigned` - (Integer) The virtual cores that are assigned to the instance.
 - `volumes` - (List of strings) The list of volume IDs that are attached to the instance.
+
+  **Notes** Ibmi software licenses for IBMi virtual server instances -- only for IBMi instances
+- `pi_ibmi_css` - (Boolean) IBMi Cloud Storage Solution.
+- `pi_ibmi_pha` - (Boolean) IBMi Power High Availability.
+- `pi_ibmi_rds` - (Boolean) IBMi Rational Dev Studio.
+- `pi_ibmi_rds_users` - (Integer) IBMi Rational Dev Studio Number of User Licenses.
```
Comment (yussufsh) [inline website/docs/d/pi_instance.html.markdown:83]: Arrange this in alphabetical order as you did in the resource doc.

File: ibm/service/power/data_source_ibm_pi_instance.go (hunk 3 above)
Comment (yussufsh) [reply ibm/service/power/data_source_ibm_pi_instance.go]: IMO it is easy to read while working with the resource to differentiate between arg and attr. I will leave it to you.

File: website/docs/d/pi_instance.html.markdown (hunk 5 above)
Comment (ismirlia) [reply website/docs/d/pi_instance.html.markdown]: Updated documentation

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 6]
```diff
@@ -772,8 +801,37 @@ func resourceIBMPIInstanceUpdate(ctx context.Context, d *schema.ResourceData, me
...
+			}
+		}
+
+		sl := &models.SoftwareLicenses{}
+		sl.IbmiCSS = flex.PtrToBool(d.Get(Arg_IBMiCSS).(bool))
+		sl.IbmiPHA = flex.PtrToBool(d.Get(Arg_IBMiPHA).(bool))
+		ibmrdsUsers := d.Get(Arg_IBMiRDSUsers).(int)
+		if ibmrdsUsers < 0 {
+			return diag.Errorf("request with IBMi Rational Dev Studio property requires IBMi Rational Dev Studio number of users")
+		}
```
Comment (yussufsh) [inline ibm/service/power/resource_ibm_pi_instance.go:820]: Not a requirement but we should be using ValidateFunc
https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#validatefunc

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 7]
```diff
@@ -1356,6 +1464,31 @@ func createPVMInstance(d *schema.ResourceData, client *st.IBMPIInstanceClient, i
...
+		if ibmiCSS, ok := d.GetOk(Arg_IBMiCSS); ok {
+			sl.IbmiCSS = flex.PtrToBool(ibmiCSS.(bool))
+		}
+		if ibmiPHA, ok := d.GetOk(Arg_IBMiPHA); ok {
+			sl.IbmiPHA = flex.PtrToBool(ibmiPHA.(bool))
+		}
+		if ibmrdsUsers, ok := d.GetOk(Arg_IBMiRDSUsers); ok {
+			if ibmrdsUsers.(int) < 0 {
+				return nil, fmt.Errorf("request with IBMi Rational Dev Studio property requires IBMi Rational Dev Studio number of users")
+			}
```
Comment (yussufsh) [inline ibm/service/power/resource_ibm_pi_instance.go:1485]: Same as https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5082/files#r1497298024

File: ibm/service/power/resource_ibm_pi_instance.go (hunk 6 above)
Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_instance.go:820]: That's probablt a bit more than I want to change for this PR. Probably mark it as an enhancement.

**Review by yussufsh — APPROVED** [review]
lgtm

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 8]
```diff
@@ -353,6 +353,28 @@ func ResourceIBMPIInstance() *schema.Resource {
...
+			},
+			Arg_IBMiPHA: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "IBMi Power High Availability",
+			},
+			Attr_IBMiRDS: {
+				Type:        schema.TypeBool,
+				Optional:    false,
+				Required:    false,
```
[the remark this answers is not in the document]
Comment (yussufsh) [reply ibm/service/power/resource_ibm_pi_instance.go:369]: `Computed cannot be used with Required.`

**Review by michaelkad — APPROVED** [review]

**Comment by ismirlia** [comment]
@hkantare Please merge this if there are no more comments.

**Review by hkantare — APPROVED** [review]

**Pull request merged**